### PR TITLE
Update documentation about question types

### PR DESCRIPTION
**Pull request by chrisroos**
Describe 'User input', 'Options', 'Validation' and 'Response' for each
question type. I've split these onto multiple lines as I think it's
easier to read.

### CODE REVIEW

File: doc/smart-answer-flows.md [hunk 1]
```diff
@@ -29,14 +29,61 @@ lib
...
+  * User input: Enter any text.
+  * Validation (depends on the `parse` option):
+    * `Integer`: Must be a number.
+    * `:to_i`: Should be a number but non numbers are valid.
+    * `Float`: Must be a number.
+    * `:to_f`: Should be a number but non numbers are valid.
+    * `<anything-else>`: No validation.
+  * Response (depends on the `parse` option):
+    * `Integer`: Integer.
+    * `:to_i`: Integer.
```
Comment (floehopper) [inline doc/smart-answer-flows.md]: Non-numeric # => `0`

File: doc/smart-answer-flows.md [hunk 2]
```diff
@@ -29,14 +29,61 @@ lib
...
+    * `Float`: Float.
+    * `:to_f`: Float.
```
Comment (floehopper) [inline doc/smart-answer-flows.md]: Non-numeric # => `0.0`

**Comment by floehopper** [comment]
LGTM :+1:

File: doc/smart-answer-flows.md (hunk 1 above)
Comment (chrisroos) [inline doc/smart-answer-flows.md]: I've added this note in 9063a231989fdf2c4d9169c13723eb99d44fd1d1.

File: doc/smart-answer-flows.md (hunk 2 above)
Comment (chrisroos) [inline doc/smart-answer-flows.md]: I've added this note in 9063a231989fdf2c4d9169c13723eb99d44fd1d1.

**Comment by chrisroos** [comment]
Merged manually in aaad80991eebe194f5e827dab13c3f74e7c7cb26 using:

    $ git merge --no-ff improve-documentation-about-question-types  -m "Merge pull request #2124"

**Pull request closed without merging**